src/core/client/stream/tabs/Profile/ChangeUsername/ChangeUsernameContainer.tsx

```diff
@@ -50,14 +51,29 @@ const FREQUENCYSCALED = reduceSeconds(ALLOWED_USERNAME_CHANGE_FREQUENCY, [
 
 interface Props {
   viewer: ViewerData;
+  settings: SettingsData;
 }
 
 interface FormProps {
   username: string;
   usernameConfirm: string;
 }
 
-const ChangeUsernameContainer: FunctionComponent<Props> = ({ viewer }) => {
+type AuthIntegrations = "local" | "sso" | "google" | "facebook" | "oidc";
+
+function enabledAuthenticationIntegrations(tenant: SettingsData): string[] {
```

Comment:
ok so I have this function on both components (change email and change username) and I'm not sure how to share this logic without things getting fairly tangled with types and imports. The function needs to know what a tenant is, but `SettingsData` is specific to the fragment requested in each component, so I'm not sure how to abstract this out. On the server side there are generic types shared across the codebase but on the client the types are all defined based on the fragments, and these two components don't have the same fragments. I was thinking that I'd need to have a separate component that just does the show/not show logic that requests the specific fragment it needs? but then it seems like overkill to have to pass that fragment and its type around between this component and the next one up. Any thoughts?